src/utils/ValidationUtils.js

```diff
@@ -47,18 +49,40 @@ function isValidUUIDArray(uuids :any[]) :boolean {
 
 function isValidDataPrimitive(value :any) :boolean {
 
-  return isBoolean(value) || isFinite(value) || isPlainObject(value) || isString(value);
+  return isBoolean(value) || isFinite(value) || isString(value);
 }
 
 function isValidDataPrimitiveArray(values :any) :boolean {
 
   return validateNonEmptyArray(values, (value :any) => isValidDataPrimitive(value));
 }
 
+function isObjectOrMap(value :any) :boolean %checks {
+  return isPlainObject(value) || Map.isMap(value);
+}
+
+function validateArrayOrListWith(values :any, predicate :Function) :boolean {
+  if (Array.isArray(values)) {
+    for (let index = 0; index < values.length; index += 1) {
+      if (!predicate(values[index])) {
+        return false;
+      }
+    }
```

Comment:
Can't you do `.every(predicate);` here as well?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every

Reply:
oh shit. indeed.